test/suites/integration/conversion_test.go

```diff
@@ -0,0 +1,132 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	karpv1beta1 "sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	karptest "sigs.k8s.io/karpenter/pkg/test"
+
+	v1 "github.com/aws/karpenter-provider-aws/pkg/apis/v1"
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-provider-aws/pkg/test"
+)
+
+var _ = Describe("Conversion Webhooks", func() {
+	Context("NodePool", func() {
+		It("should not update a metadata generation when the same resource is applied for the v1beta1 APIs", func() {
+			// created v1beta1 resource
+			storedv1beta1NodePool := &karpv1beta1.NodePool{
+				ObjectMeta: karptest.ObjectMeta(),
+				Spec: karpv1beta1.NodePoolSpec{
+					Template: karpv1beta1.NodeClaimTemplate{
+						Spec: karpv1beta1.NodeClaimSpec{
```

Comment:
Should we add in the fields here that we know were problematic? e.g. resources, NillableDuration, kubelet resources

Reply:
I was waiting to add the resources and NilableDuration fields after the fix was put merged, but I can also add them in now. What were the kubelet resources fields that were problematic? I validated that we had removed all fields that used the resource.Quantity.